src/System.Private.CoreLib/src/System/StartupHookProvider.cs

```diff
@@ -106,12 +105,12 @@ private static void CallStartupHook(string assemblyPath)
             if (wrongSignature)
             {
                 throw new ArgumentException(SR.Format(SR.Argument_InvalidStartupHookSignature,
-                                                      startupHookTypeName + Type.Delimiter + initializeMethodName,
+                                                      StartupHookTypeName + Type.Delimiter + InitializeMethodName,
                                                       assemblyPath));
             }
 
             Debug.Assert(initializeMethod != null &&
-                         initializeMethod.IsPublic &&
+                         (initializeMethod.IsPublic || initializeMethod.IsPrivate) &&
```

Comment:
This is not covering all non-public methods. Non-public methods include protected, etc. I think it is best to delete this part of the assert.